File: frontend/src/api/endpoints/notifications.ts
/** Client de /api/v1/notifications/*. */
import api from "@/api/client";

export type NotificationKind =
  | "appointment_confirmed"
  | "wo_status_changed"
  | "parts_request"
  | "qa_pending"
  | "delivery_ready"
  | "system";

export interface NotificationItem {
  id: string;
  user_id: string;
  kind: NotificationKind;
  title: string;
  body: string;
  link_url: string | null;
  read_at: string | null;
  branch_id: string | null;
  entity_type: string | null;
  entity_id: string | null;
  created_at: string;
  updated_at: string | null;
}

export interface NotificationListResponse {
  items: NotificationItem[];
  total: number;
  unread: number;
}

export interface UnreadCount {
  count: number;
}

export const notificationsApi = {
  list: (params: { unread_only?: boolean; limit?: number; skip?: number } = {}) =>
    api
      .get<NotificationListResponse>("/v1/notifications", { params })
      .then((r) => r.data),

  unreadCount: () =>
    api.get<UnreadCount>("/v1/notifications/unread-count").then((r) => r.data),

  markRead: (id: string) =>
    api.post<NotificationItem>(`/v1/notifications/${id}/read`).then((r) => r.data),

  markAllRead: () =>
    api.post<{ updated: number }>("/v1/notifications/mark-all-read").then((r) => r.data),
};

File: app/schemas/notifications.py
"""Pydantic schemas for in-app notifications."""
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict

from app.models.notifications import NotificationKind


class NotificationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: str
    kind: NotificationKind
    title: str
    body: str
    link_url: Optional[str] = None
    read_at: Optional[datetime] = None
    branch_id: Optional[str] = None
    entity_type: Optional[str] = None
    entity_id: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None


class NotificationListResponse(BaseModel):
    items: list[NotificationRead]
    total: int
    unread: int


class UnreadCount(BaseModel):
    count: int
